pkg/controller/migplan/migplan_controller.go

```diff
@@ -532,87 +529,100 @@ func (r ReconcileMigPlan) waitForMigAnalyticsReady(plan *migapi.MigPlan) (*migap
 	migAnalytics := &migapi.MigAnalyticList{}
 	err := r.List(context.TODO(), k8sclient.MatchingLabels(map[string]string{MigPlan: plan.Name}), migAnalytics)
 	if err != nil {
-		return nil, err
+		return nil, liberr.Wrap(err)
 	}
-	for _, migAnalytic := range migAnalytics.Items {
+	for i := range migAnalytics.Items {
+		migAnalytic := &migAnalytics.Items[i]
 		if migAnalytic.Spec.AnalyzeExtendedPVCapacity == true {
-			if migAnalytic.Status.IsReady() {
-				return &migAnalytic, nil
+			if migAnalytic.Status.IsReady() && !migAnalytic.Spec.Refresh {
+				return migAnalytic, nil
 			}
 			if time.Now().Sub(migAnalytic.CreationTimestamp.Time) > migAnalyticsTimeout {
 				plan.Status.SetCondition(migapi.Condition{
-					Type:     IntelligentPVResizingDisabled,
+					Type:     PVResizingDisabled,
 					Status:   True,
 					Category: Warn,
-					Message:  "MigAnalytics did not complete on time, going ahead with the migration without offering pvc resize function",
+					Reason:   NotDone,
+					Message:  "Failed to gather reliable PV usage data from the source cluster, PV resizing will be disabled.",
 				})
+				plan.Status.DeleteCondition(PvCapacityAdjustmentRequired)
 			}
 		}
 
 	}
 	return nil, nil
 }
 
-func (r ReconcileMigPlan) processExtendedPVCapacity(plan *migapi.MigPlan, analytic *migapi.MigAnalytic) {
+// processProposedPVCapacities reads miganalytic status to find proposed capacities of volumes
+func (r ReconcileMigPlan) processProposedPVCapacities(plan *migapi.MigPlan, analytic *migapi.MigAnalytic) {
+	pvResizingRequiredVolumes := []string{}
 	plan.Spec.PersistentVolumes.BeginPvStaging()
 	for i := range plan.Spec.PersistentVolumes.List {
 		planVol := &plan.Spec.PersistentVolumes.List[i]
 		for _, analyticNS := range analytic.Status.Analytics.Namespaces {
 			if planVol.PVC.Namespace == analyticNS.Namespace {
 				for _, analyticNSVol := range analyticNS.PersistentVolumes {
 					if planVol.PVC.Name == analyticNSVol.Name {
-
-						// If plan volume capacity is less than analytic volume's proposed capacity, set confirmed to False
-						if planVol.Capacity.Cmp(analyticNSVol.ProposedCapacity) < 0 {
-							planVol.CapacityConfirmed = false
-						}
-
-						// If new proposed capacity is greater than original capacity, set confirmed to False
+						// If new proposed capacity is greater than previous proposed capacity, set confirmed to False
 						if planVol.ProposedCapacity.Cmp(analyticNSVol.ProposedCapacity) > 0 {
 							planVol.CapacityConfirmed = false
 						}
+						if analyticNSVol.Comment != miganalytic.VolumeAdjustmentNoOp {
```

Comment:
What is a comment in this context?

Reply:
`Comment` is a string set by MigAnalytic controller about the proposed size. If it takes no action on the capacities, a no-op string is set.

Reply:
@pranavgaikwad would `ProposedSize` be a more specific name for this then? Or is this a multi-use field.

Reply:
@djwhatle `ProposedCapacity` is a field we are actually using to store the 'value' of corrected size and `Comment` is a 'type' of correction applied from the criteria described here: https://github.com/pranavgaikwad/mig-operator/blob/pvResizeDocs/docs/usage/IntelligentPVResizing.md#criteria

Reply:
Gotcha. Sounds good. Marking as resolved.